Code/GraphMol/Substruct/SubstructMatch.cpp

```diff
@@ -35,6 +35,20 @@
 
 namespace RDKit {
 namespace detail {
+
+namespace {
+bool hasChiralLabel(const Atom *at) {
+  return at->getChiralTag() == Atom::CHI_TETRAHEDRAL_CW ||
+         at->getChiralTag() == Atom::CHI_TETRAHEDRAL_CCW;
+}
+
+//! Decide if an atom comes from a query that describes a mol fragment
+bool isFragmentAtom(const Atom *at) {
+  unsigned explicitDegree = at->getDegree() + at->getNumExplicitHs();
+  return explicitDegree < 4 && at->getImplicitValence() == -1;
```

Comment:
Wouldn't this be just as valid if you used:
```suggestion
  return explicitDegree < 4 && at->getImplicitValence() <= 0;
```
A use case where this could happen: I could imagine manually constructing a query molecule by starting from a normal molecule and removing one atom from a chiral center and then re-sanitizing (making sure that chiral center has the noImplicit flag set). I think with your current logic this would end up matching anything when doing chiral searches. That doesn't seem right.

Reply:
Yeah. I need to reconsider all the "fragment" treatment in this PR. From the beginning of it, it has been the part I have been less confident on.